pdns/ws-auth.cc

```diff
@@ -2255,6 +2255,14 @@ static void apiServerCacheFlush(HttpRequest* req, HttpResponse* resp) {
 
   DNSName canon = apiNameToDNSName(req->getvars["domain"]);
 
+  SOAData sd;
+  UeberBackend B;
+  if (g_zoneCache.isEnabled() && B.getSOAUncached(canon, sd)) {
+    // zone exists (uncached), add/update it in the zone cache.
```

Comment:
this only checks that a SOA exists in `records`. It does not check that the domain is in `domains`. If it is not, itl'l fall out of the zone cache in ~30 seconds, which is confusing. Can you add that check?

Reply:
Right, I shall switch this to use B.getAuth(... cachedOk=false) really!

Reply:
> Right, I shall switch this to use B.getAuth(... cachedOk=false) really!

Hmm, does that actually hit `domains`?

Reply:
Hmm, no. getDomainInfo it is, then.